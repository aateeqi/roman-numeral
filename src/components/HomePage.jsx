import { navigate } from "@reach/router";
import React, { useState } from "react";


export const Home = () => {
  const [numeral, setNumeral] = useState(-1);

  const handleSubmit = (e) => {
    e.preventDefault();
    if (isNaN(numeral) || numeral < 0 || numeral >= 4000) {
      alert(`Invalid input: ${numeral}`);
    } else {
      navigate(`/romannumeral?query=${numeral}`);
    }
  };

  return (
    <div>
      <h1>Hello, Welcome to the greatest App</h1>
      <div>
        <form onSubmit={handleSubmit}>
          <label>
            Enter Number:
            <input
              type="text"
              name="query"
              onChange={(event) => setNumeral(event.target.value)}
            />
          </label>
          <input type="submit" value="Submit" />
        </form>
      </div>
    </div>
  );
};
