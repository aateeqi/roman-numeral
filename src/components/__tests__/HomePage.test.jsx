import React from "react";
import renderer from "react-test-renderer";
import { Home } from "../HomePage";

test("renders correctly", () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
