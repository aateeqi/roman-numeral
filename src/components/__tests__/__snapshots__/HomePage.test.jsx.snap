// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Hello, Welcome to the greatest App
  </h1>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label>
        Enter Number:
        <input
          name="query"
          onChange={[Function]}
          type="text"
        />
      </label>
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
